habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PyenvAndPoetrySetup.java

```
@@ -52,59 +52,63 @@ public PyenvAndPoetrySetup(String pythonVersion, File baseDir, boolean rewriteLo
-                        "Ensure your PATH includes 'pyenv init -' expected content " +
-                        "OR do not configure habushu to use 'pyenv' to manage the Python version!");
-            }
+    private String validateAndConfigurePythonViaPyenv(String currentPythonVersion, File patchInstallScript) throws MojoExecutionException {
+        PyenvCommandHelper pyenvHelper = createPyenvCommandHelper();
+        try {
+            currentPythonVersion = pyenvHelper.getCurrentPythonVersion();
+        } catch (Exception e) {
+            log.info("Failed to find current python version. Attempting to install it now.");
+        }
+        if (!pythonVersion.equals(currentPythonVersion)) {
+            pyenvHelper.updatePythonVersion(pythonVersion, patchInstallScript);
+            currentPythonVersion = pyenvHelper.getCurrentPythonVersion();
+        }
 
-            log.debug("pyenv already installed");
+        // Check for misconfigured pyenv that looks right, but is actually not "taking" due to missing PATH setup:
+        PythonVersionHelper pythonVersionHelper = new PythonVersionHelper(baseDir, pythonVersion);
+        String postPyenvActivatedPythonVersion = pythonVersionHelper.getCurrentPythonVersion();
+        List<String> pyenvMisconfiguration  = new ArrayList<>();
+        if (!pythonVersion.equals(postPyenvActivatedPythonVersion)) {
+            pyenvMisconfiguration.add(String.format("Expected 'pyenv' to set Python to %s but instead found %s!",
+                    pythonVersion, postPyenvActivatedPythonVersion));
+                    pyenvMisconfiguration.add("'pyenv' is installed, but not configured correctly.  " +
+                    "Ensure your PATH includes 'pyenv init -' expected content " +
+                    "OR do not configure habushu to use 'pyenv' to manage the Python version!");
+            throw new MojoExecutionException(StringUtils.join(System.lineSeparator(), pyenvMisconfiguration, System.lineSeparator()));
```

I: lnstead of passing these error messages back to the abstract method via `missingRequiredToolMsgs`, we're now throwing an exception here to fail fast because `pyenv` isn't configured correctly.

---

S: We initially did it via `missingRequiredToolMsgs` because people get aggravated when they fix one tool and find out another is missing.  So, if we can do both at once, it makes for a better onboarding experience.  I'm probably be inclined to keep it that way.